core/src/main/java/org/apache/iceberg/jdbc/JdbcCatalog.java

```diff
@@ -245,13 +286,18 @@ public List<TableIdentifier> listTables(Namespace namespace) {
         row ->
             JdbcUtil.stringToTableIdentifier(
                 row.getString(JdbcUtil.TABLE_NAMESPACE), row.getString(JdbcUtil.TABLE_NAME)),
-        JdbcUtil.LIST_TABLES_SQL,
+        JdbcUtil.LIST_TABLE_SQL,
         catalogName,
-        JdbcUtil.namespaceToString(namespace));
+        JdbcUtil.namespaceToString(namespace),
+        "TABLE");
```

Comment:
We should probably pull the `TABLE` and `VIEW` literals into static variables (e.g. `TABLE_RECORD_TYPE` and `VIEW_RECORD_TYPE`) as opposed to relying on strings here.  It's not particularly clear why we're passing this value as a param.

Reply:
Good idea, let me do that.